api/storage_service.cc

```diff
@@ -721,7 +721,7 @@ static
 json::json_return_type
 rest_get_natural_endpoints(http_context& ctx, sharded<service::storage_service>& ss, const_req req) {
         auto keyspace = validate_keyspace(ctx, req);
-        auto res = ss.local().get_natural_endpoints(keyspace, req.get_query_param("cf"), req.get_query_param("key"));
+        auto res = ss.local().get_natural_endpoints(keyspace, req.get_query_param("cf"), req.get_query_param("key"), req.get_query_param("key_delimiter"));
```

Comment:
This makes the key_delimiter a de-facto required parameter, breaking all current clients of the API.
If the parameter is not required, you need to handle the case where it is not provided.